src/rgw/rgw_sal_motr.cc

```diff
@@ -2474,24 +2660,37 @@ int MotrObject::read_mobj(const DoutPrefixProvider* dpp, int64_t off, int64_t en
     rc = m0_obj_op(this->mobj, M0_OC_READ, &ext, &buf, &attr, 0, 0, &op);
     ldpp_dout(dpp, 20) << __func__  << ": init read op rc = " << rc << dendl;
     if (rc != 0) {
-      ldpp_dout(dpp, 0) << __func__ << ": read failed during m0_obj_op, rc = " << rc << dendl;
+      ldpp_dout(dpp, 0) << __func__ << ": read failed during m0_obj_op, rc =" << rc << dendl;
+      ADDB(RGW_ADDB_REQUEST_ID, addb_logger.get_id(), 
+	   RGW_ADDB_FUNC_READ_MOBJ,
+	   RGW_ADDB_PHASE_ERROR);
       goto out;
     }
+    ADDB(RGW_ADDB_REQUEST_TO_MOTR_ID, addb_logger.get_id(), m0_sm_id_get(&op->op_sm));
     m0_op_launch(&op, 1);
     rc = m0_op_wait(op, M0_BITS(M0_OS_FAILED, M0_OS_STABLE), M0_TIME_NEVER) ?:
          m0_rc(op);
     m0_op_fini(op);
     m0_op_free(op);
     if (rc != 0) {
-      ldpp_dout(dpp, 0) << __func__ << ": read failed, m0_op_wait rc = " << rc << dendl;
+      ldpp_dout(dpp, 0) << __func__ << ": read failed, m0_op_wait rc =" << rc << dendl;
+      ADDB(RGW_ADDB_REQUEST_ID, addb_logger.get_id(), 
+	   RGW_ADDB_FUNC_READ_MOBJ,
+	   RGW_ADDB_PHASE_ERROR);
       goto out;
     }
     // Call `cb` to process returned data.
     ldpp_dout(dpp, 20) << __func__  << " call cb to process data" << dendl;
     cb->handle_data(bl, bloff, actual);
     bloff = 0;
+
+    addb_logger.set_id(req_id);
```

Comment:
Why is this needed?

Reply:
The reason why this at first glance not necessary call is needed relates to the nature of multithreading in RGW.
You might have seen that `addb_logger` object resides in thread local storage. It is needed because it allows to associate each thread with particular unique ID that will be logged into ADDB stob. It also allows us to avoid modification of RGW sources and map a RGW request ID to subsequent motr client request IDs - to plot request sequential diagrams, histograms, etc. 
So, ideally, once request ID is assigned to thread local storage (i.e., addb_logger), it might be used from any place of the code. But it is not true for RGW threading model. RGW is based on asio threads. and its engine may (and actually does) change thread working on a particular request. Assume req0 is processed by thread0, and and thread0 is assigned with req0 ID. Next, `cb->handle_data()` is executed, and req0 is now executed by thread1, while thread0 is returned back to thread pool. And it means we have to re-assign req ID to thread1 so it can be dumped correctly.